accepted/single-file/design.md

```diff
@@ -66,225 +32,254 @@ Publishing to a single file can be triggered by adding the following property to
+* Tests to ensure cross-platform publishing of single-file bundles.
+* Manual end-to-end testing on real world apps
 
 #### Measurements
 
-Measure publish size and run-time (first run, subsequent runs) for HelloWorld (framework-dependent and self-contained), Roslyn and MusicStore.
+Measure publish size and startup time for a few real-world apps.
 
 #### Telemetry
 
 Collect telemetry for single-file published apps with respect to parameters such as:
 
 * Framework-dependent vs self-contained apps.
 * Whether the apps are Pure managed apps, ready-to run compiled apps, or have native dependencies.
-* Embedding of additional/data files, and use of file access API.
+* Embedding of additional/data files.
 
 ## Further Work
 
-#### Bundler Optimizations
-
-Since all the files of an app published as a single-file live together, we can perform the following optimizations
-
-- R2R compile the app and all of its dependent assemblies in a single version-bubble
-
-  Single-file apps compiled cross-platform may have this optimization disabled until the ready-to-run compiler  (`crossgen`) supports cross-compilation. 
-
-- Investigate whether collectively signing the files in an assembly saves space for certificates.
-
-#### Compression
-
-Currently the bundler does not compress the contents embedded at the end of the host binary. 
-Compressing the bundled files and meta-data can significantly reduce the size of the single-file output (by about 30%-50% as determined by prototyping).
-
-#### Single-file Plugins
-
-The above design should be extended to seamlessly support single-file publish for plugins.
-
-- The bundler tool will mostly work as-is, regardless of whether we publish an application or class-lib. The binary blob with dependencies can be appended to both native and managed binaries.
-- For host/runtime support, the options are:
-  - Implement plugins using existing infrastructure. For example: Take control of assembly/native binary loads via existing `AssemblyLoadContext` callbacks and events. Extract the files embedded within the single-file plugin using the `GetFileStream()` API and load them on demand.
-  - Have new API to load a single-file plugin, for example: `AssemblyLoadContext.LoadWithEmbeddedDependencies()`.
-
-#### VS Integration
-
-Developers should be able to use the feature easily from Visual Studio. The feature will start with text based support -- by explicitly setting the `PublishSingleFile` property in the project file. In future, we may provide a single-file publish-profile, or other UI triggers.
+* **Compression**: Currently the bundler does not compress the contents embedded at the end of the host binary.  Compressing the bundled files and meta-data can significantly reduce the size of the single-file output (by about 30%-50% as determined by prototyping). 
+* **Single-file Plugins** Extended the above design to seamlessly support single-file publish for plugins.
 
 ## User Experience
```

Comment:
I think it should be at the start of the doc. The start of the doc should explain what we are building. The implementation details should be after that.

Reply:
OK I'll change that, thanks